### DISCUSSION TITLE
AutoGG&Play

### DISCUSSION THREAD

**Discussion by Az-eeeee** [XiaoHeiPa/LiquidLunar#29] [Ideas]
import net.minecraft.network.play.server.S02PacketChat

@ModuleInfo(name = "AutoGG", category = ModuleCategory.MISC)
object AutoGG : Module() {

    private val winCheckValue = TextValue("WinCheck", "Winner")
    private val delayValue = IntegerValue("Delay", 1000, 1000, 5000)
    private val ggMessageValue = TextValue("GGMessage", "GG!")

    private var winning = false
    private val timer = MSTimer()
    override fun onEnable() {
        timer.reset()
        winning = false
    }

    @EventTarget
    fun onWorld(event: WorldEvent) {
        winning = false
        timer.reset()
    }

    @EventTarget
    fun onPacket(event: PacketEvent) {
        val packet = event.packet
        if(packet is S02PacketChat) {
            val message = packet.chatComponent.unformattedText

            if(message.contains(winCheckValue.get()) && !message.contains(":")) {
                winning = true
            }
        }
    }
**SkidderMC/FDPClient**

import net.minecraft.network.play.client.*
import net.minecraft.network.play.server.S02PacketChat
import net.minecraft.network.play.server.S2DPacketOpenWindow
import net.minecraft.network.play.server.S2FPacketSetSlot
import net.minecraft.util.IChatComponent
import java.util.*
import kotlin.concurrent.schedule

@EventTarget
    fun onPacket(event: PacketEvent) {
        val packet = event.packet

        when (modeValue.get().toLowerCase()) {
         "hypixel" -> {
                if (clickState == 1 && packet is S2DPacketOpenWindow) {
                    event.cancelEvent()
                }
            }
        }
        if (packet is S2FPacketSetSlot) {
            val item = packet.func_149174_e() ?: return
            val windowId = packet.func_149175_c()
            val slot = packet.func_149173_d()
            val itemName = item.unlocalizedName
            val displayName = item.displayName

            when (modeValue.get().toLowerCase()) {
               "blocksmc", "hypixel" -> {
                    if (clickState == 0 && windowId == 0 && slot == 43 && itemName.contains("paper", ignoreCase = true)) {
                        queueAutoPlay {
                            mc.netHandler.addToSendQueue(C09PacketHeldItemChange(7))
                            repeat(2) {
                                mc.netHandler.addToSendQueue(C08PacketPlayerBlockPlacement(item))
                            }
                        }
                        clickState = 1
                    }
                    if (modeValue.equals("hypixel") && clickState == 1 && windowId != 0 && itemName.equals("item.fireworks", ignoreCase = true)) {
                        mc.netHandler.addToSendQueue(C0EPacketClickWindow(windowId, slot, 0, 0, item, 1919))
                        mc.netHandler.addToSendQueue(C0DPacketCloseWindow(windowId))
                    }
                }
            }
        } else if (packet is S02PacketChat) {
            val text = packet.chatComponent.unformattedText
            when (modeValue.get().toLowerCase()) {
              "hypixel" -> {
                    fun process(component: IChatComponent) {
                        val value = component.chatStyle.chatClickEvent?.value
                        if (value != null && value.startsWith("/play", true)) {
                            mc.thePlayer.sendChatMessage("/ac [Noteless]GG Hacked By Dimples#1337")
                            queueAutoPlay {
                                mc.thePlayer.sendChatMessage(value)
                            }
                        }
                        component.siblings.forEach {
                            process(it)
                        }
                    }
                    process(packet.chatComponent)
               }
           }
       }
}

private fun queueAutoPlay(runnable: () -> Unit) {
        if (queued) {
            return
        }
        queued = true
        if (this.state) {
            Timer().schedule(delayValue.get().toLong() * 1000) {
                if (state) {
                    runnable()
                }
            }
    @EventTarget
    fun onWorld(event: WorldEvent) {
        clicking = false
        clickState = 0
        queued = false
    }

    override val tag: String
        get() = modeValue.get()

    override fun handleEvents() = true
}

**Dimples#1337-AutoGG**

**Comment by cubewhy** [comment]
updated
